core/database_util.go

```diff
@@ -138,6 +189,168 @@ func GetBody(db ethdb.Database, hash common.Hash) *types.Body {
 	return body
 }
 
+// WriteBlockAddTxIndexes writes atx-indexes for a given block.
+func WriteBlockAddTxIndexes(indexDb ethdb.Database, block *types.Block) error {
+	batch := indexDb.NewBatch()
+	if _, err := putBlockAddrTxsToBatch(batch, block); err != nil {
+		return err
+	}
+	return batch.Write()
+}
+
+// putBlockAddrTxsToBatch formats and puts keys for a given block to a db Batch.
+// Batch can be written afterward if no errors, ie. batch.Write()
+func putBlockAddrTxsToBatch(putBatch ethdb.Batch, block *types.Block) (txsCount int, err error) {
+	for _, tx := range block.Transactions() {
+		txsCount++
+
+		from, err := tx.From()
+		if err != nil {
+			return txsCount, err
+		}
+		// Note that len 8 because uint64 guaranteed <= 8 bytes.
+		bn := make([]byte, 8)
+		binary.LittleEndian.PutUint64(bn, block.NumberU64())
+
+		if err := putBatch.Put(formatAddrTxBytesIndex(from.Bytes(), bn, []byte("f"), tx.Hash().Bytes()), nil); err != nil {
+			return txsCount, err
+		}
+
+		to := tx.To()
+		if to == nil || to.IsEmpty() {
```

Comment:
it might be interesting to allow `to.IsEmpty()`, since doing so would let you grab just contracts (deployed by anyone) on the chain by passing the empty hash address `0x000...`

```shell
⟠ curl -X POST --data '{"jsonrpc":"2.0","method":"debug_getAddressTransactions","params":["0x0000...", 5200000, 0, ""],"id":67}' :8545
```